config/http_config.go

```diff
@@ -356,8 +363,8 @@ func (c *HTTPClientConfig) Validate() error {
 		if strings.ToLower(c.Authorization.Type) == "basic" {
 			return fmt.Errorf(`authorization type cannot be set to "basic", use "basic_auth" instead`)
 		}
-		if c.BasicAuth != nil || c.OAuth2 != nil {
-			return fmt.Errorf("at most one of basic_auth, oauth2 & authorization must be configured")
+		if c.BasicAuth != nil || c.OAuth2 != nil || c.DigestAuth != nil {
+			return fmt.Errorf("at most one of basic_auth, digest_auth, oauth2 & authorization must be configured")
```

Comment:
these scattered checks to ensure only one type of auth is enabled bug me: more places to go wrong.
Maybe refactor them to a function that ensures only one type of auth is configured, and drop them in as needed.

```
auths_enabled := []bool{
  c.BasicAuth != nil,
  c.OAuth2 != nil,
  c.DigestAuth != nil,
  c.Authorization != nil,
}
if len(auths_enabled) > 1 {
  //  error
}
```

Reply:
Agree! I found these checks a bot strange as well and would vote for refactoring them (willing to do that!). Only wondering if this refactoring shall be included here or rather as part of another PR, so we can get this one merged preferably soon? Let me know what you think is better.

Reply:
i'm not the maintainer of this codebase, but if I had a choice, I'd say should be a separate PR that merges before this one.

Reply:
Who is a maintainer and could eventually be responsible for merging my PR?

Reply:
?

Reply:
@muety  you want @SuperQ or @mmorel-35

Reply:
@SuperQ @mmorel-35 Could someone of you possibly have a look at this, please?

Reply:
Would really like to either get this merged or rejected soon.

Reply:
I'm sorry but I'm not maintainer here, just a contributor

Reply:
Okay, sorry to bother you then. Who can I tag here to finally (!) get this finished?!

Reply:
?????

Reply:
This took me like half a day of work. Sorry, but I find it quite disrespectful that none of the maintainer even only bothers to reply.